### ISSUE TITLE
Clippy lint `redundant_closure_for_method_calls` incorrectly handles raw identifiers when applying fixes

### ISSUE THREAD

**Issue by FractalFir** [rust-lang/rust-clippy#12599]
### Summary

When applying fixes to a codebase containing raw identifiers, code produced by `clippy fix` includes a path with a keyword.

In particular, it converted:
```rust
 tpe.1.all_types() .filter_map(|tpe| tpe.dotnet_refs())
```
into 
```rust
tpe.1.all_types() .filter_map(super::r#type::r#type::Type::dotnet_refs)
```
when it should have converted it into:
```rust
tpe.1.all_types() .filter_map(super::type::type::Type::dotnet_refs)
```
the clippy lint message also contained an invalid path:
```
warning: redundant closure
   --> src/assembly.rs:678:33
    |
678 |                     .filter_map(|tpe| tpe.dotnet_refs())
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^ help: replace the closure with the method itself: `super::type::type::Type::dotnet_refs`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure_for_method_calls
```
The path should be `super::r#type::r#type::Type::dotnet_refs` instead.

### Reproducer

This short snippet is enough to trigger the issue:
```rust
mod r#type {
    pub struct Type;
    impl Type {
        pub fn f(&self) -> u8 {
            0
        }
    }
}
fn main() {
    let types = [r#type::Type];
    let _fs: Vec<_> = types.iter().map(|tpe| tpe.f()).collect();
}
```

### Version

```text
rustc 1.79.0-nightly (c9f8f3438 2024-03-27)
binary: rustc
commit-hash: c9f8f3438a8134a413aa5d4903e0196e44e37bbc
commit-date: 2024-03-27
host: x86_64-unknown-linux-gnu
release: 1.79.0-nightly
LLVM version: 18.1.2
```
